### PR TITLE
Renew certificate changes

### PR DESCRIPTION
**Pull request by nithyal1**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 1]
```diff
@@ -2711,6 +2721,213 @@ private boolean areCertificateApproleInputsValid(CertificateApprole certificateA
...
+     * @param certManagerLogin
+     * @return
+     * @throws Exception
+     */
+    private CertificateData getLatestCertificate(String certName, String accessToken) throws Exception {
+        CertificateData certificateData=null; 
+        int containerId = getTargetSystemGroupId(SSLCertType.valueOf("PRIVATE_SINGLE_SAN"));
+        String findCertificateEndpoint = "/certmanager/findCertificate";
+        String targetEndpoint = findCertificate.replace("certname", String.valueOf(certName)).replace("cid", String.valueOf(containerId));
+        CertResponse response = reqProcessor.processCert(findCertificateEndpoint, "", accessToken, getCertmanagerEndPoint(targetEndpoint));
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:2895]: can we check the response status code if available?

File: tvaultui/src/app/Features/admin/admin.controller.js [hunk 2]
```diff
@@ -146,7 +146,7 @@
             $scope.isUserSearchLoading = false;
             $scope.isOwnerSelected = false;
             setTargetSystemServiceList("No target system selected", []);
-            $scope.certificateData.certificates = [];
+            $scope.certificateData.certificates = [];
```
Comment (sarathmohan4) [inline tvaultui/src/app/Features/admin/admin.controller.js:149]: new white spaces are introduced

File: tvaultui/src/app/Features/admin/admin.controller.js [hunk 3]
```diff
@@ -1899,6 +1910,44 @@
                 console.log(e);
             }
         };
+        
+        
+        $scope.renewCertificate = function(certificateDetails){
```
Comment (sarathmohan4) [inline tvaultui/src/app/Features/admin/admin.controller.js:1915]: Can we add a confirmation popup before reniew?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java (hunk 1 above)
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:2896]: Done

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 4]
```diff
@@ -2711,6 +2721,213 @@ private boolean areCertificateApproleInputsValid(CertificateApprole certificateA
...
+
+		Map<String, String> metaDataParams = new HashMap<String, String>();
+
+		String endPoint = certificateName;
+		String _path = SSLCertificateConstants.SSL_CERT_PATH + "/" + endPoint;
+		Response response = null;
+		try {
+			if (userDetails.isAdmin()) {
+				response = reqProcessor.process("/read", "{\"path\":\"" + _path + "\"}", token);
+			} else {
```
[the remark this answers is not in the document]
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:2757]: Done

File: tvaultui/src/app/Features/admin/admin.controller.js (hunk 3 above)
Comment (nithyal1) [reply tvaultui/src/app/Features/admin/admin.controller.js]: Added

**Pull request merged**